nni/compression/pytorch/quantization_speedup/frontend_to_onnx.py

```diff
@@ -124,21 +110,24 @@ def torch_to_onnx(model, config, input_shape, model_path, input_names, output_na
     name2index = {}
     if config is not None:
         for i, name in enumerate(config.keys()):
-            index2name[i] = name
-            name2index[name] = i
+            # i*2 to avoid the graph optimization of onnx on constant folding / elimination
+            # for example, multiply 1 will be removed from the graph
+            index2name[i * 2] = name
+            name2index[name] = i * 2
     for name, module in model.named_modules():
         if config is not None and name in config:
             assert type(module) in support_op
             wrapper_module = LayernameModuleWrapper(module, name2index[name])
-            _setattr(model, name, wrapper_module)
+            set_nested_attr(model, name, wrapper_module)
         elif type(module) in support_op:
             wrapper_module = LayernameModuleWrapper(module, -1)
-            _setattr(model, name, wrapper_module)
+            set_nested_attr(model, name, wrapper_module)
     # Convert torch model to onnx model and save it in model_path
+    device = torch.device('cpu')
     dummy_input = torch.randn(input_shape)
```

Comment:
consider a case， if there are multiple inputs, such as bert's input contains multiple items, input_ids, attention_masks, ..., how to generate dummy_input at this time? In addition, in some cases, such as the input_ids of BERT, each element is an integer and has a range. At this time, using torch.randn() will cause certain problems.

Reply:
these are very good cases. leave it to future prs.